test/integration/resource.ts

```diff
@@ -0,0 +1,41 @@
+import {expect} from 'chai';
+import {Ketting, NeverCache} from '../../src';
+
+describe('Using resources', async () => {
+  const ketting = new Ketting('http://localhost:3000/hal1.json');
+
+  it('concurrent get on the same resource should do 2 distinct HTTP calls if request headers are different', async () => {
+    ketting.cache = new NeverCache();
+
+    const resource = await ketting.follow('collection');
+
+    const [firstState, secondState] =
+      await Promise.all([
+        resource.get({headers: {'delay-in-ms': '2', 'foo': 'bar'}}),
+        resource.get({headers: {'foo': 'baz'}})
```

Comment:
The 2 requests are different because of the `delay-in-ms`. `delay-in-ms` should not impact the equality between the 2 requests.
Give them the same delay and use `prefer` instead of `foo`.

Reply:
Done